GitUI/GitUIExtensions.cs

```diff
@@ -86,20 +86,30 @@ public static Task ViewChanges(this FileViewer diffViewer, IList<GitRevision> re
 
         public static Task ViewChanges(this FileViewer diffViewer, string firstRevision, string secondRevision, GitItemStatus file, string defaultText)
         {
-            return diffViewer.ViewPatch(() =>
+            if (firstRevision == null)
             {
-                if (firstRevision == null && secondRevision.IsNotNullOrWhitespace())
+                // The previous commit does not exist, nothing to compare with
+                if (file.TreeGuid.IsNullOrEmpty())
                 {
-                    // The previous commit does not exist, nothing to compare with
-                    if (file.TreeGuid.IsNullOrEmpty())
-                        diffViewer.ViewGitItemRevision(file.Name, secondRevision);
-                    else
-                        diffViewer.ViewGitItem(file.Name, file.TreeGuid);
+                    if (secondRevision.IsNullOrWhiteSpace())
+                    {
+                        throw new ArgumentException(nameof(secondRevision));
```

Comment:
I would like to return defaultText in 2.51, suggest investigation #4586 is done first.

Reply:
I don't understand your comment. Could you please clarify it?

Reply:
The use of null as first or second revision is not consistent. In some situations the revisions are null when comparing. I would like to review that handling, suggest to not merge to 2.51 (or use the initial commit in 2.51).